Automattic/wp-calypso
# Site Domains Page: Add All Domains Section

Related to https://github.com/Automattic/dotcom-forge/issues/2871

## Proposed Changes

* Add a link to the site domains page that redirects the user to the all domains management dashboard

## Screenshots
<img width="1496" alt="Screenshot 2023-07-12 at 5 29 55 PM" src="https://github.com/Automattic/wp-calypso/assets/5414230/293604bf-33ef-4149-927a-4e33ea2a00ba">

<img width="1496" alt="Screenshot 2023-07-12 at 5 31 42 PM" src="https://github.com/Automattic/wp-calypso/assets/5414230/c02d1f09-4faa-4568-8bc2-bca1668bd8a2">

## Testing Instructions



* Check out branch, `yarn` and `yarn start`
* Navigate to the sites domain page for a site without custom domains Ex. http://calypso.localhost:3000/domains/manage/tests12312.wordpress.com
* Verify that the manage all domains button appears at the bottom of the page
* Verify that the button redirects the user to the all domains page
* Navigate to the sites domain page for a site with custom domains Ex. http://calypso.localhost:3000/domains/manage/test-newsletter-0005.blog
* Verify that the manage all domains page button appears at the bottom of the page
* Verify that the button redirects the user to the all domains page

## Pre-merge Checklist
